src/blockdata/constants.rs

```diff
@@ -128,6 +128,21 @@ pub fn genesis_block(network: Network) -> Block {
                 txdata: txdata
             }
         }
+        Network::Signet => {
+            // TODO: somehow obtain the block script and generate coinbase tx from it
```

Comment:
Does this mean it's currently incorrect?

Reply:
No, it means the block hash is not derived automatically, but hard coded. There's a related discussion at https://github.com/bitcoin/bitcoin/pull/16411#issuecomment-577999888 which suggests that we use a hard coded genesis block for all signet instances -- right now, signet derives the genesis block from the challenge (i.e. scriptPubKey of the blocks).

Reply:
Actually, I take that back. I'm actually unsure about this one. Will have to verify this one.

Reply:
I've pushed a to-squash commit that properly derives the signet genesis based on the block script. Note that https://github.com/bitcoin/bitcoin/pull/16411#issuecomment-577999888 is still up in the air, but currently keeping things as is.

Reply:
So currently all signets will have the same genesis hash with the signet0 pubkey in it? Cool. 

ACK if that's what Core is doing.

Reply:
In practice, it is possible to specify a custom block script which would result in a different genesis hash, but recent discussion indicates we will usually have only one signet at a time, most of the time.